apps/web/components/apps/AppPage.tsx

```diff
@@ -70,22 +73,63 @@ export const AppPage = ({
   dependencies,
   concurrentMeetings,
   paid,
-  dirName,
 }: AppPageProps) => {
   const { t, i18n } = useLocale();
+  const router = useRouter();
   const searchParams = useCompatSearchParams();
 
   const hasDescriptionItems = descriptionItems && descriptionItems.length > 0;
 
-  const mutation = useAddAppMutation(null, {
-    onSuccess: (data) => {
-      if (data?.setupPending) return;
-      showToast(t("app_successfully_installed"), "success");
-    },
-    onError: (error) => {
-      if (error instanceof Error) showToast(error.message || t("app_could_not_be_installed"), "error");
-    },
-  });
+  const mutation = useAddAppMutation(null);
+
+  const [isLoading, setIsLoading] = useState<boolean>(mutation.isPending);
```

Comment:
Why not use mutation.isPending directly?

Reply:
the reason for this was that it was taking some time for the next page to load as we have some big queries in next page, so the loader would turn off but the page wouldn't change

see https://github.com/calcom/cal.com/pull/15193#pullrequestreview-2094125667
and https://github.com/calcom/cal.com/pull/15193#issuecomment-2147531336

Reply:
TBH this sounds like a hack more than a proper solution. I'm ok with compromising this change but we need to acknowledge this in a code comment at least and make an issue to refactor.

Reply:
yeah, it's somewhat hacky 
will add a comment for now, and think of some other way to handle this

Reply:
pushed a better solution 

https://github.com/calcom/cal.com/pull/15193/commits/b2f478cdb68ab0c5caf30c581e10547548bafbbd and https://github.com/calcom/cal.com/pull/15193/commits/69378039ec5df39cd7e74d39bf4c3fd148ac537b

@joeauyeung @zomars